### PR TITLE
fix: boolean operator for boolean operands

### PR DESCRIPTION
**Pull request by priyasiddharth**
also innocuous clang-format changes

### CODE REVIEW

**Comment by priyasiddharth** [comment]
@jorgenavas PTAL. Thanks

**Review by agurfinkel — COMMENTED** [review]
there are too many formatting changes. what line has changed

File: lib/Clam/SeaDsaHeapAbstraction.cc [hunk 1]
```diff
@@ -433,28 +432,29 @@ void SeaDsaHeapAbstraction::initialize(const llvm::Module &M) {
       // Keep track of inconsistent memory regions (i.e., regions on
       // which caller and callee disagree)
       for (unsigned i = 0, e = readsC.size(); i < e; i++) {
-        readsB[i] = readsB[i] & readsC[i].second;
-	if (!readsC[i].second) {
-	  CLAM_WARNING("Caller and callee disagree on some read region type\n"
-		       << "Callsite=" << *CI << "\n"
-		       << "Caller=" << *CS.getCaller() << "\n");
-	}
+        readsB[i] = readsB[i] && readsC[i].second;
```
Comment (priyasiddharth) [inline lib/Clam/SeaDsaHeapAbstraction.cc:435]: this is an important change

File: lib/Clam/SeaDsaHeapAbstraction.cc [hunk 2]
```diff
@@ -433,28 +432,29 @@ void SeaDsaHeapAbstraction::initialize(const llvm::Module &M) {
...
+        }
       }
       for (unsigned i = 0, e = modsC.size(); i < e; i++) {
-        modsB[i] = modsB[i] & modsC[i].second;
-	if (!modsC[i].second) {
-	  CLAM_WARNING("Caller and callee disagree on some modified region type\n"
-		       << "Callsite=" << *CI << "\n"
-		       << "Caller=" << *CS.getCaller() << "\n");
-	}	
+        modsB[i] = modsB[i] && modsC[i].second;
```
Comment (priyasiddharth) [inline lib/Clam/SeaDsaHeapAbstraction.cc:443]: this is an important change

Comment (agurfinkel) [reply lib/Clam/SeaDsaHeapAbstraction.cc:443]: @caballa please double check. Crab overrides `&` so sometimes bit-operation is the right one, no matter what clang says

Comment (priyasiddharth) [reply lib/Clam/SeaDsaHeapAbstraction.cc:443]: if this is intended then the specific warning to be silenced is `-Wbool-operation` using [pragma directive](https://clang.llvm.org/docs/UsersManual.html#controlling-diagnostics-via-pragmas)

Comment (caballa) [reply lib/Clam/SeaDsaHeapAbstraction.cc:443]: Clang is right here

**Comment by caballa** [comment]
Fixed in commit [cd60c15](https://github.com/seahorn/clam/commit/cd60c15fe50f8d4dcf4a42a5add9f82147625a02)

**Pull request closed without merging**